.evergreen/auth_aws/aws_e2e_ec2.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Verify the AWS IAM EC2 hosted auth works
+ */
+load("lib/aws_e2e_lib.js");
+
+(function() {
+"use strict";
+
+// This varies based on hosting EC2 as the account id and role name can vary
+const AWS_ACCOUNT_ARN = "arn:aws:sts::557821124784:assumed-role/authtest_instance_profile_role/*";
+
+function assignInstanceProfile() {
+    const config = readSetupJson();
+
+    const env = {
+        AWS_ACCESS_KEY_ID: config["iam_auth_ec2_instance_account"],
+        AWS_SECRET_ACCESS_KEY: config["iam_auth_ec2_instance_secret_access_key"],
+    };
+
+    const instanceProfileName = config["iam_auth_ec2_instance_profile"];
+    const python_command = getPython3Binary() +
+        ` -u lib/aws_assign_instance_profile.py --instance_profile_arn=${instanceProfileName}`;
```

Comment:
FYI, This script is failing sometimes in evergreen because of excessive retries.

I am working around in with: https://jira.mongodb.org/browse/SERVER-46023. I am not ready to commit yet though.